src/platform/common/stm32/config_flash.c

```diff
@@ -0,0 +1,473 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "platform.h"
+#include "drivers/system.h"
+#include "config/config_streamer.h"
+
+#if defined(CONFIG_IN_FLASH)
+
+#if defined(STM32F745xx) || defined(STM32F746xx) || defined(STM32F765xx)
+/*
+Sector 0    0x08000000 - 0x08007FFF 32 Kbytes
+Sector 1    0x08008000 - 0x0800FFFF 32 Kbytes
+Sector 2    0x08010000 - 0x08017FFF 32 Kbytes
+Sector 3    0x08018000 - 0x0801FFFF 32 Kbytes
+Sector 4    0x08020000 - 0x0803FFFF 128 Kbytes
+Sector 5    0x08040000 - 0x0807FFFF 256 Kbytes
+Sector 6    0x08080000 - 0x080BFFFF 256 Kbytes
+Sector 7    0x080C0000 - 0x080FFFFF 256 Kbytes
+
+F7X5XI device with 2M flash
+Sector 8    0x08100000 - 0x0813FFFF 256 Kbytes
+Sector 9    0x08140000 - 0x0817FFFF 256 Kbytes
+Sector 10   0x08180000 - 0x081BFFFF 256 Kbytes
+Sector 11   0x081C0000 - 0x081FFFFF 256 Kbytes
+*/
+
+uint32_t getFLASHSectorForEEPROM(void)
+{
+    if ((uint32_t)&__config_start <= 0x08007FFF)
```

Comment:
THis would be much cleaner as array + `for` loop:

```
uint32_t flashBase = 0x08000000;
struct {
   uint32_t sector;
   uint32_t top;
} flashSectorMap[] = {
{FLASH_SECTOR_0, 0x08007FFF}, // better use start of next sector, 0x08008000
FLASH_SECTOR_1, ...

Reply:
Agree - further PR required once devices will be available for testing